### ISSUE TITLE
UI notes

### ISSUE THREAD

**Issue by meetar** [polygon-city/building-database#92]
Login
- [ ] Set focus to user field on page load?

Add Building
- [ ] Make a note somewhere on the page of accepted file types
- [ ] Do we accept KMZ files? They aren't selectable in my Chrome upload dialog but I'm not sure why
- [ ] Progress indicator during upload? Maybe an animated ellipsis "..." or percentage completed in the "Uploading..." button? eg: "Uploading… 55%"

Set Building Location
- [ ] Possible to set zoom of map based on size of model? Small models are almost invisible
- [ ] Modal manipulators? Maybe [click/rightclick] changes [scale/rotation] when building is selected, and [location/camera orbit] when building is unselected, as per current behavior?
- [ ] Disable map capturing arrow key events when dropdowns have focus
- [ ] Lots of "undefined"s in location suggestions – is this typical of Nominatum or something adjustable on our end?
- [ ] Faster map zooming on mouse wheel scrolling?

Link to OSM
- [ ] Is the "node type" information still necessary? If it is, can't we get that from the selected node? Regardless, it'd be nice to eliminate the dropdown
- [ ] The popover bubble that appears when you select a node often covers a lot of map space… can we move that to the side somehow? Maybe to a separate pane below the map?

Building information page
- [ ] Zoom or scale tile or model to model extents? Small models are almost invisible
- [ ] In many places in the world, the Stamen toner map is completely blank – of course it's less likely that anyone would place a building in those places, but still. It's not particularly urgent, and I don't have clear idea what would solve this one, apart from a different tileset – the page load, we could start zoomed out and zoom in to the building location, to set context… there's always the good old Economist-style mini-world-globe with a positioning dot… And if in the future all the maps are interactive, the user could just zoom out to get context if they want it.

**Comment by robhawkes** [comment]
e5d1fe8 fixes the undefined errors after setting a location.

**Comment by robhawkes** [comment]
1c87151 (in #69) introduces an interim fix for model scaling on the building page. It's not perfect but it looks a lot better than before, especially for smaller models.

**Comment by robhawkes** [comment]
078f070 adds a fix that fits the `/add/location` view to the building.

**Comment by robhawkes** [comment]
@meetar: "Set focus to user field on page load?" – this already happens for me as there's an `autofocus` attribute set on the username field. Do you not see this?

**Comment by robhawkes** [comment]
> Is the "node type" information still necessary? If it is, can't we get that from the selected node? Regardless, it'd be nice to eliminate the dropdown

If we want the ability to manually input the ID (eg. to account for something we may have missed) then we need both as I believe the IDs can be duplicated across various types (nodes, ways, relations). If not, then we can remove both fields entirely and re-design the UI to focus purely on selecting objects we define. If the latter, I suggest it becomes a v2 feature.

**Comment by robhawkes** [comment]
> Do we accept KMZ files? They aren't selectable in my Chrome upload dialog but I'm not sure why

My gut says this is a good idea, though it sets an expectation that I'm unsure if we're able to meet right now. We can support KMZ from a technical standpoint quite easily (it's just a zip), and if it only contains a single model than we can do that too. The problem lies with a KMZ (or zip for that matter) that contains more than 1 model – we don't support multi-model uploads yet. With a zip this isn't so much a problem as I kind of cheat by selecting only one model from the zip (assuming they're duplicates). With KMZ it's expected that the models within are separate, so we can't take that approach.

In short, KMZ is definitely something we support. In my mind it should be a v2 feature and be added alongside multi-model / batch uploads.

**Comment by robhawkes** [comment]
b9358c9 adds basic upload progress % on the add building button.

**Comment by robhawkes** [comment]
b93abad reduces the number of tags shown in the Overpass popups which makes them much less imposing. This isn't a full fix but it certainly makes things better for the time being.
